### DISCUSSION TITLE
add support for specifying a file's language using a comment in the file itself

### DISCUSSION THREAD

**Discussion by bonds** [helix-editor/helix#7555] [Ideas] [answered]
I have scripts written in various languages. They start with a shebang line and usually the rest is code in the chosen language. When I open a file named 'mycoolscript' in helix, helix doesn't know what language the file is in.

It would be nice if there was some way to annotate the file to let helix know what language it's in. For example, in vim I can add a line like this: ```# vi: ft=haskell``` and vim is able to set the correct language when the file loads.

**Comment by pascalkuthe** [comment]
> They start with a shebang line 

helix already supports filetye selections using shebangs. Your language is likely just not commonly used with a shebang so the detection fails

**Comment by bonds** [comment, marked as answer]
Ah! Thanks for the clue! While it's not the same mechanism and I was thinking about, at least for Haskell I can just add a ~/.config/helix/languages.toml with:

```
[[language]]
name = "haskell"
shebangs = ["cabal", "stack", "runghc", "runhaskell"]
```

And my scripts now load into helix with the correct 'haskell' filetype. The default languages.toml does not have a shebang line for Haskell, so it doesn't work by default, like you said.
